ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,279 @@
+- **Optional Continuous Sync**: ArgoCD (when available) continuously propagates changes from repo to namespace
+- **Manual Alternative**: Without optional services, teams manually manage namespace creation and GitOps synchronization
+- **1:1 Mapping**: Each GitOps repo maps to exactly one namespace
+- **Immutable Names**: Namespace names are declared in the repo and cannot change
+- **Single Deployment Target**: All resources from one repo deploy to one namespace
+- **No Cross-Namespace Resources**: Each team's resources are isolated within their own namespace
+
+
+### CI/Validation Design
+
+- Introduce a Konflux CLI validator (or GitHub Action) for local + CI use
+- Validates schema, references, build logic statically
+- Same validator embedded in VS Code plugin
+- Optional: Plugin lints against live cluster (e.g., available build agents)
+
+### UI Changes
+
+**Preserve:**
+- Dashboard
+- Build and release logs
+- Operational actions: triggering builds/tests, starting manual releases, pipeline management
+
+**Remove/Demote:**
+- UI forms for configuring components, applications, and other declarative resources
+
+## Consequences
+
+### Positive Consequences
+
+1. **Unified Developer Experience**: Single Git-centric flow eliminates context switching between UI and GitOps
+2. **Faster Onboarding**: Local validation reduces time-to-merge for onboarding PRs (enhanced with optional services)
+3. **Clear Tenant Model**: 1:1 mapping between GitOps repos and namespaces provides clear ownership and boundaries
+4. **Improved Team Autonomy**: Multi-repo model allows teams to own their GitOps repositories and corresponding namespaces
+5. **Better IDE Integration**: Leverages familiar VS Code tooling and workflows
+6. **Flexible Deployment Options**: Optional services allow deployment scaling from lightweight setups to full-featured production environments
+7. **Resource Efficiency**: Minimal footprint option suitable for lower-spec machines and testing environments
+8. **Optional GitOps Automation**: When ArgoCD is deployed, provides industry-standard continuous deployment with automatic drift detection and reconciliation
+9. **Maintained GitOps Benefits**: Preserves disaster recovery, change history, and automation capabilities (enhanced with optional services)
+
+### Negative Consequences
+
+1. **Tool Dependency**: Requires VS Code and plugin installation for optimal experience
+2. **Learning Curve**: Developers must adapt to new plugin-based workflow
+3. **Development Overhead**: Requires building and maintaining VS Code plugin and optional services
+4. **Manual Operations**: Without optional services, teams must manually manage GitOps operations, reducing automation benefits
+5. **Migration Complexity**: Existing users must migrate from current UI/GitOps hybrid approach
+6. **Configuration Complexity**: Teams must choose between manual setup (simple but limited) vs. optional services (full-featured but more complex)
+
```

Comment:
Currently, some controllers use spec and annotations to communicate. For example, when one creates ImageRepository, there could be annotation that define for which Component this IR is created and operator automatically updates the required Component fields.
Another example with Image Controller is credentials rotation. A user needs to set a spec field to `true` and when the operator refresh the tokens, the field is returned to `false`. Will this play well with pure GitOps approach?